lib/linter.js

```diff
@@ -74,12 +74,19 @@ module.exports = function (cwd, args, text) {
 
     return JSON.stringify(fileConfig, null, '  ');
   }
+  if (currentOptions.fixToStdout && !stdin) {
+    return 'The --fix-to-stdout option must be used with --stdin.'
+      + '\n# exit 1';
+  }
   var report;
   if (stdin && text) {
     report = engine.executeOnText(text, currentOptions.stdinFilename);
   } else {
     report = engine.executeOnFiles(files);
   }
+  if (currentOptions.fixToStdout) {
+    return report.results[0].output || text;
```

Comment:
Related to above, there's an edge case here if a user passes `--stdin`, no text, and more than one file. I should probably at least check for `text` in the conditional, but that would introduce another edge case... if a user lint fixed a blank file, you'd want it to return a blank file (maybe with a trailing newline? 😄)